lib/teleterm/daemon/daemon.go

```diff
@@ -139,15 +139,23 @@ func (s *Service) CreateGateway(ctx context.Context, params clusters.CreateGatew
 }
 
 // createGateway assumes that mu is already held by a public method.
-func (s *Service) createGateway(ctx context.Context, params clusters.CreateGatewayParams) (*gateway.Gateway, error) {
+func (s *Service) createGateway(ctx context.Context, params CreateGatewayParams) (*gateway.Gateway, error) {
 	cluster, err := s.ResolveCluster(params.TargetURI)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
 	cliCommandProvider := clusters.NewDbcmdCLICommandProvider(s.Storage)
 
-	gateway, err := cluster.CreateGateway(ctx, cliCommandProvider, params)
+	clusterCreateGatewayParams := clusters.CreateGatewayParams{
+		TargetURI:             params.TargetURI,
+		TargetUser:            params.TargetUser,
+		TargetSubresourceName: params.TargetSubresourceName,
+		LocalPort:             params.LocalPort,
+		CLICommandProvider:    cliCommandProvider,
+	}
```

Comment:
@marcoandredinis This commit moves `CLICommandProvider` within `clusters.CreateGatewayParams`. Let me know what you think.

I'm not sure if it was worth it though. 🤔 I don't know how to deal with converting between `daemon.CreateGatewayParams` (which doesn't have the `CLICommandProvider` field) to `clusters.CreateGatewayParams` without explicitly listing all fields like this.

Reply:
I think it's fine :+1: